tests/test_study.py

```diff
@@ -712,7 +712,9 @@ def test_append_trial(storage_mode):
         study = optuna.create_study(storage=storage)
         assert len(study.trials) == 0
 
-        study._append_trial(value=0.8)
+        trial_id = study._append_trial(value=0.8) == 0
+        if isinstance(study._storage, optuna.storages.InMemoryStorage):
+            assert trial_id == 0
```

Comment:
We can also check a trial id on RDBStorage like the following:

```python
        if isinstance(study._storage, optuna.storages.InMemoryStorage):
            assert study._append_trial(value=0.8) == 0
        else:
            assert study._append_trial(value=0.8) == 1
```

But I only checked trial_id when using InMemoryStorage. As far as I know, it is an anti-pattern to depend on the primary key (e.g. Django's default test case doesn't reset the sequences for some performance reasons. [refs django.test.TransactionTestCase.reset_sequences](https://docs.djangoproject.com/en/3.0/topics/testing/advanced/#django.test.TransactionTestCase.reset_sequences)).

Reply:
I think that since there is only one trial we can check the correctness of the trial id as follows:
```python
assert study.trials[0]._trial_id == trial_id
```

Reply:
It's a good idea 👍